### PR TITLE
Minimal reproducible code draft

### PR DESCRIPTION
**Pull request by kaijagahm**
Full draft of the minimal reproducible code lesson. Includes full lesson structure. 

Notes:

- I dumped a giant script at the beginning that includes the code in the researcher narrative up through the part that was assigned to this episode. I realize this is overwhelming and gross, and we can discuss changes to it.
     - @peterlaurin and I discussed this today and thought that it was generally a good approach, rather than trying to work the code in gradually, since we're not actually trying to teach coding in this episode. I still have some reservations, but let's roll with it for now.
     - I wanted to use code folding, and tried to, but it didn't work. Asking in Carpentries Slack about how to fix this. If we can't get it to work, we should include this as a separate script that users have to download.
     - I do think it's important to have them all working with the exact same script version at least for this part. 

- Critically, this is still missing content for the exercises, and that's largely because the script itself does not contain enough errors. Need to introduce some errors into the script, especially in the later portions (modeling, plotting, answering research questions) so that we can then refer to those in the lesson.
- Time estimates are probably way off.

- @peterlaurin and @xortizross there are some spots indicated here for you to add references back to your own episodes

### CODE REVIEW

**Review by peterlaurin — APPROVED** [review]
Looks good -- I did not check all the renv stuff but it seems to be passing the checks so I think we can just let it be?

File: episodes/5-minimal-reproducible-code-draft.Rmd [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+-   I want to make my code minimal, but where do I even start?
+-   How do I make non-reproducible code reproducible?
+-   How do I tell whether a code snippet is reproducible or not?
+:::
+
+::: objectives
+-   Identify the step that is generating the error
+-   Implement a stepwise approach to make minimal code
+-   Edit a piece of code to make it reproducible
+-   Evaluate whether a piece of code is reproducible as is or not. If not, identify what is missing.
```
Comment (xortizross) [inline episodes/5-minimal-reproducible-code-draft.Rmd:18]: Again, I feel like this should go before the previous one... (first figure out whether it is reproducible, then fix it to make it so)

File: episodes/5-minimal-reproducible-code-draft.Rmd [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+---
+
+::: questions
+-   Which part of my code is causing an error message or an incorrect result?
+-   I want to make my code minimal, but where do I even start?
+-   How do I make non-reproducible code reproducible?
+-   How do I tell whether a code snippet is reproducible or not?
+:::
+
+::: objectives
```
Comment (xortizross) [inline episodes/5-minimal-reproducible-code-draft.Rmd:14]: Love these changes!

File: episodes/5-minimal-reproducible-code-draft.Rmd [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+-   Identify the step that is generating the error
+-   Implement a stepwise approach to make minimal code
+-   Edit a piece of code to make it reproducible
+-   Evaluate whether a piece of code is reproducible as is or not. If not, identify what is missing.
+:::
+
+In the last episode, we focused in on how to make minimal datasets that would reproduce a target piece of code, but we didn't talk much about the code itself. In each example, we already had a piece of code that needed a dataset. But how do you know which part of your code is the problem and should be focused on?
+
+At this point, we as researchers have been exploring our kangaroo rat data for a while and making several plots. Now we know how many kangaroo rats we can expect to catch, and we can move on to our second research question: Do the rodent exclusion plots actually working at keeping the target species out of certain areas?
```
Comment (xortizross) [inline episodes/5-minimal-reproducible-code-draft.Rmd:23]: check grammar

File: episodes/6-asking-your-question.md [hunk 4]
```diff
@@ -6,20 +6,24 @@ exercises: 2
...
+- How can I make sure my minimal reproducible example will actually run correctly for someone else?
 - How can I easily share a reproducible example with a mentor or helper, or online?
 - How do I ask a good question?
 
 ::::::::::::::::::::::::::::::::::::::::::::::::
 
 ::::::::::::::::::::::::::::::::::::: objectives
 
-- To do
+- Use the reprex package to support making reproducible examples.
```
Comment (xortizross) [inline episodes/6-asking-your-question.md:17]: ?

File: episodes/6-asking-your-question.md [hunk 5]
```diff
@@ -6,20 +6,24 @@ exercises: 2
 
 :::::::::::::::::::::::::::::::::::::: questions 
 
+- How can I make sure my minimal reproducible example will actually run correctly for someone else?
 - How can I easily share a reproducible example with a mentor or helper, or online?
 - How do I ask a good question?
 
 ::::::::::::::::::::::::::::::::::::::::::::::::
 
 ::::::::::::::::::::::::::::::::::::: objectives
```
Comment (xortizross) [inline episodes/6-asking-your-question.md:15]: Maybe "Describe the reprex package and its benefits "

File: episodes/6-asking-your-question.md [hunk 6]
```diff
@@ -6,20 +6,24 @@ exercises: 2
...
+- Understand the benefits and drawbacks of different places to get help.
+- Have a road map to follow when posting a question to make sure it's a good question.
 
 ::::::::::::::::::::::::::::::::::::::::::::::::
 
-
 ::::::::::::::::::::::::::::::::::::: keypoints
 
 - The {reprex} package makes it easy to format and share your reproducible examples.
+- Following a certain set of steps will make your questions clearer and likelier to get answered.
```
Comment (xortizross) [inline episodes/6-asking-your-question.md:27]: I would actually list the steps here but I understand we are not there yet

File: renv/profiles/lesson-requirements/renv/settings.json
Comment (xortizross) [inline renv/profiles/lesson-requirements/renv/settings.json:1]: ??

File: renv/profiles/lesson-requirements/renv/.gitignore
Comment (xortizross) [inline renv/profiles/lesson-requirements/renv/.gitignore:1]: ??

File: renv/profiles/lesson-requirements/renv.lock
Comment (xortizross) [inline renv/profiles/lesson-requirements/renv.lock:1]: ??

File: renv/activate.R
Comment (xortizross) [inline renv/activate.R:1]: ??

File: episodes/5-minimal-reproducible-code-draft.Rmd [hunk 7]
```diff
@@ -0,0 +1,166 @@
+---
+title: "Minimal Reproducible Code"
+teaching: 10
+exercises: 2
+---
+
+::: questions
+-   Which part of my code is causing an error message or an incorrect result?
```
[the remark this answers is not in the document]
Comment (kaijagahm) [reply episodes/5-minimal-reproducible-code-draft.Rmd:8]: Ooh yes I like this.

**Pull request merged**